private/bufpkg/bufanalysis/bufanalysis.go

```diff
@@ -64,6 +64,11 @@ var (
 		FormatJSON: "json",
 		FormatMSVS: "msvs",
 	}
+	formatToPrinter = map[Format]printer{
+		FormatText: printerFunc(textPrinter),
+		FormatJSON: printerFunc(jsonPrinter),
+		FormatMSVS: printerFunc(msvsPrinter),
+	}
```

Comment:
Instead of having this be a global, we could rework `FormatFileAnnotation` with the `switch` statement to instead operate on the entire set of file annotations, `[]FileAnnotation`?

The concern is for the `junit` implementation, where we'll need all the `FileAnnotations` to print, so the print logic would need to be one level up from the single `FileAnnotation` unit, so by reworking the logic such that we handle the entire slice of `[]FileAnnotation` given the `formatString` provided to `PrintAllFileAnnotations` should be sufficient.

Reply:
I think we can remove `FormatFileAnnotation` altogether, it is only being used in tests. 

IMO separating the printing part from `FileAnnotation` is useful as the formats can grow. Instead of modifying the `FileAnnotation` interface we can just add a new `Format` and `printer`. 

The `FileAnnotation` can have a default format in the form of `Stringer` interface (used in tests of `bufimagebuild`).

Not that it is major concern here, but this is slightly faster than using `FormatFileAnnotation` to return a `string` as we can reuse the `bytes.Buffer`

Reply:
So I think that we probably want to prevent `Format` from becoming more than just the enum for managing format types (which this current design adheres to), however, we should also avoid this global here if it can be handled by a `switch`. We should also avoid attaching the `printer` interface to the `Format`, even though they are 1:1 since that logic can be encapsulated into a single `Print` function that takes in the format and file annotations (without defining a new structure). I agree with the decision to probably get rid of `FormatFileAnnotation`, since for the `junit` output in particular, we will need to print all `FileAnnotation`s together, so formatting each file annotation separately doesn't really make much sense anymore.